### PR TITLE
Don't translate char codes when platform,encoding isn't (3,0)

### LINKED ISSUE #17808: PDF Document does not display text correctly when rendered in pdf.js
Attach (recommended) or Link to PDF file here:

Configuration:
- Web browser and its version: Mozilla any version
- Operating system and its version: Any Os, Tested on windows
- PDF.js version: 4.0.379
- Is a browser extension:

Steps to reproduce the problem:
1. Open the Attached PDF in https://mozilla.github.io/pdf.js/web/viewer.html
2. We see broken text in rendered document

What is the expected behavior? (add screenshot)
Text should be displayed correctly 
What went wrong? (add screenshot)

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

**Comment by calixteman** [comment]
@Snuffleupagus, I'm not very sure of what I'm doing... so don't worry to take the ownership if you've a better idea.

**Comment by Snuffleupagus** [comment]
There's a typo in the commit message: platorm -> platform

Also, for this sort of change, running all tests *before* asking for feedback/review seems like a good idea :-)

File: src/core/fonts.js [hunk 1]
```diff
@@ -2969,15 +2966,17 @@ class Font {
         // cmap.
         for (const mapping of cmapMappings) {
           let charCode = mapping.charCode;
-          if (
-            cmapPlatformId === 3 &&
-            charCode >= 0xf000 &&
-            charCode <= 0xf0ff
-          ) {
+          if (charCode >= 0xf000 && charCode <= 0xf0ff) {
```
Comment (Snuffleupagus) [inline src/core/fonts.js:2969]: Can't we simply reduce the *entire patch* to just the following changes instead, which is less overall duplication?
```suggestion
          if (
            cmapPlatformId === 3 &&
            cmapEncodingId === 0 &&
            charCode >= 0xf000 &&
            charCode <= 0xf0ff
          ) {
```

Comment (calixteman) [reply src/core/fonts.js:2969]: I initially did that but I wasn't happy to add an extra a potentially useless check in the loop and I found it's more consistent to have the platformID, encodingId checks at the same level as the other one. Finally I think the code is more clear like this.

**Review by Snuffleupagus — APPROVED** [review]
While test coverage should be pretty good for this part of the font code, we probably need to be prepared for regressions from this change nonetheless.

r=me, thank you!

**Comment by marco-c** [comment]
If the change is risky, I'd suggest waiting a bit before landing to make uplifts to 125 easier.

**Comment by Snuffleupagus** [comment]
> If the change is risky, I'd suggest waiting a bit before landing to make uplifts to 125 easier.

Given that https://bugzilla.mozilla.org/show_bug.cgi?id=1889118 was recently landed, should we try merging this PR now?

**Comment by marco-c** [comment]
Yes, I think we're safe to land it now!

**Pull request merged**